dev/arxiv_category_converter.py

```
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+"""
+ArXiv category code to user-friendly name converter.
+Called by arxiv_fetch.py to convert category codes to readable names.
+"""
+# Standard library
+import csv
+import os
+
+# Third-party
+import yaml
+
+
+def load_category_mapping(data_dir):
+    """Load category code to label mapping from YAML file."""
+    mapping_file = os.path.join(data_dir, "arxiv_category_map.yaml")
+
+    if not os.path.exists(mapping_file):
+        return {}
+
+    try:
+        with open(mapping_file, "r") as f:
+            return yaml.safe_load(f) or {}
+    except Exception:
+        return {}
```

Hi @Goziee-git! 

I have a question about this. Why do you choose to silently swallow up the exceptions? Would it not be better to log an error/warning so the user knows the mapping wasn't loaded?

---

This script is still in development and thanks @Babi-B for the notes here. I have looked into it carefully and have implemented changes to properly log exceptions instead of silently handling them.